File: app/src/main/java/com/jawnnypoo/physicslayout/sample/github/GithubClient.java
package com.jawnnypoo.physicslayout.sample.github;


import com.jawnnypoo.physicslayout.sample.BuildConfig;

import java.util.List;

import retrofit.Callback;
import retrofit.RestAdapter;
import retrofit.http.GET;
import retrofit.http.Path;

/**
 * Created by Jawn on 6/30/2015.
 */
public class GithubClient {

    public static final String API_URL = "https://api.github.com";

    public interface GitHub {
        @GET("/repos/{owner}/{repo}/contributors")
        void contributors(
                @Path("owner") String owner,
                @Path("repo") String repo,
                Callback<List<Contributor>> callback);
    }

    private static GitHub mGithub;
    public static GitHub instance() {
        if (mGithub == null) {
            RestAdapter restAdapter = new RestAdapter.Builder()
                    .setEndpoint(API_URL)
                    .setLogLevel(BuildConfig.DEBUG ? RestAdapter.LogLevel.FULL : RestAdapter.LogLevel.NONE)
                    .build();
            mGithub = restAdapter.create(GitHub.class);
        }
        return mGithub;
    }
}

File: app/src/main/java/com/jawnnypoo/physicslayout/sample/github/GithubClient.kt
package com.jawnnypoo.physicslayout.sample.github


import com.jawnnypoo.physicslayout.sample.BuildConfig

import retrofit.Callback
import retrofit.RestAdapter
import retrofit.http.GET
import retrofit.http.Path

object GithubClient {

    private const val API_URL = "https://api.github.com"

    private val github: GitHub by lazy {
        val restAdapter = RestAdapter.Builder()
                .setEndpoint(API_URL)
                .setLogLevel(if (BuildConfig.DEBUG) RestAdapter.LogLevel.FULL else RestAdapter.LogLevel.NONE)
                .build()
        restAdapter.create(GitHub::class.java)
    }

    fun instance(): GitHub {
        return github
    }

    interface GitHub {
        @GET("/repos/{owner}/{repo}/contributors")
        fun contributors(
                @Path("owner") owner: String,
                @Path("repo") repo: String,
                callback: Callback<List<Contributor>>)
    }
}
